src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -1565,11 +1565,11 @@ void java_lang_Thread::serialize_offsets(SerializeClosure* f) {
 #endif
 
 JavaThread* java_lang_Thread::thread(oop java_thread) {
-  return reinterpret_cast<JavaThread*>(java_thread->address_field(_eetop_offset));
+  return reinterpret_cast<JavaThread*>(java_thread->address_field_acquire(_eetop_offset));
 }
 
 void java_lang_Thread::set_thread(oop java_thread, JavaThread* thread) {
-  java_thread->address_field_put(_eetop_offset, (address)thread);
+  java_thread->release_address_field_put(_eetop_offset, (address)thread);
 }
```

Comment:
There is a convention that when we have acquire/release semantics, that we make that evident in the name of the method so that it is obvious at the call-site. So these would need renaming to `thread_acquire` and `release_set_thread`.

Reply:
Or, to avoid a lot of changes, add new methods for the acquire/release variants. Really only the setting to non-null requires release semantics. Which reads need acquire semantics is much harder to analyse - but most will not due to other inherent synchronization either via Threads_lock, or at the Java level.

Reply:
I completely forgot about the `foo_acquire` and `release_set_foo` convention.
I think I'll go the route of adding `thread_acquire` and `release_set_thread` and
then will revisit each call-site to decide which to use.

I'm also going to look at splitting the `java_lang_Thread::thread_acquire` and
`java_lang_Thread::release_set_thread` changes into a sub-task. This will allow
for easier reviewing and backport decisions.